maestro-cli/src/main/java/maestro/cli/device/DeviceCreateUtil.kt

```diff
@@ -0,0 +1,145 @@
+            } else it.modelId
+        }
+
+        if (existingDeviceId != null) PrintUtils.message("Using existing device $deviceName (${existingDeviceId}).")
+        else PrintUtils.message("Attempting to create iOS simulator: $deviceName ")
+
+
+        val deviceUUID = try {
+            existingDeviceId ?: DeviceService.createIosDevice(deviceName, device, runtime).toString()
+        } catch (e: IllegalStateException) {
+            val error = e.message ?: ""
+            if (error.contains("Invalid runtime")) {
+                val msg = "Required runtime to create the simulator is not installed: $runtime\n\n" +
+                        "To install additional iOS runtimes checkout this guide:\n" +
+                        "* https://developer.apple.com/documentation/xcode/installing-additional-simulator-runtimes"
+                throw CliError(msg)
+            } else if (error.contains("Invalid device type")) {
+                throw CliError("Device type $device is either not supported or not found.")
+            } else {
+                throw CliError(error)
+            }
+        }
+
+        if (existingDeviceId == null) PrintUtils.message("Created simulator $deviceName ($deviceUUID)")
+
+        return Device.AvailableForLaunch(
+            modelId = deviceUUID,
+            description = deviceName,
+            platform = Platform.IOS
+        )
+
+    }
+
+    private fun getOrCreateAndroidDevice(version: Int?, forceCreate: Boolean): Device.AvailableForLaunch {
+        if (version !in DeviceConfigAndroid.versions) {
+            throw CliError("Provided Android version is not supported. Please use one of ${DeviceConfigAndroid.versions}")
+        }
+
+        val systemImage = DeviceConfigAndroid.systemImages[version]
+            ?: throw CliError("Provided system image is not supported. Please use one of ${DeviceConfigAndroid.versions}")
+
+        val name = DeviceConfigAndroid.generateDeviceName(version!!)
+
+        // check connected device
+        if (DeviceService.isDeviceConnected(name, Platform.ANDROID) != null) {
+            throw CliError("A device with name $name is already connected")
+        }
+
+        // existing device
+        val existingDevice =
+            if (forceCreate) null else DeviceService.isDeviceAvailableToLaunch(name, Platform.ANDROID)?.modelId
+
+        // dependencies
+        if (existingDevice == null && !DeviceService.isAndroidSystemImageInstalled(systemImage)) {
+            PrintUtils.err("The required system image $systemImage is not installed.")
+
+            PrintUtils.message("Would you like to install it? y/n")
+            val r = readlnOrNull()?.lowercase()
+            if (r == "y" || r == "yes") {
+                PrintUtils.message("Attempting to install $systemImage via Android SDK Manager...\n")
+                if (!DeviceService.installAndroidSystemImage(systemImage)) {
+                    throw CliError("Was unable to install required dependencies.")
```

Comment:
Can we provide a more detailed error message here and/or guide users to self-recover like we do below?

Reply:
Added a better message with self recovery instructions